chain/chunks/src/lib.rs

```diff
@@ -1732,10 +1734,15 @@ impl ShardsManager {
 
         Ok(())
     }
-    pub fn random_seed(&self, shard_id: ShardId) -> RngSeed {
+
+    /// Computes a deterministic random seed for given `shard_id`.
+    /// This seed is used to randomize the transaction pool.
+    /// For better security we want the seed to different in each shard.
+    /// For testing purposes we want it to be the reproducible and derived from the `self.rng_seed` and `shard_id`
+    fn random_seed(&self, shard_id: ShardId) -> RngSeed {
         let mut res = self.rng_seed.clone();
-        res[17] = shard_id as u8;
-        res[23] = (shard_id / 256) as u8;
+        res[17] = (shard_id + 777) as u8;
+        res[23] = ((shard_id + 681) / 256) as u8;
```

Comment:
this really hurts. Even if there is a need to introduce "magic" constants they should be well documented. Please explain why these numbers are chosen in the first place. If there is no good reason for them, then `0` is usually a better choice.

Reply:
FWIW, I am somewhat confused by the fact that we need to mess up with seeds at all, instead of passing an RNG instance.